cli/auth_xkey_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023-2024 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"crypto/rand"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/nats-io/nkeys"
+)
+
+func TestSealOpen(t *testing.T) {
+	// Create two pairs of xkeys
+	ef := rand.Reader
+	p1, err := nkeys.CreateCurveKeysWithRand(ef)
+	if err != nil {
+		t.Error("Failed to create key")
+		t.FailNow()
+	}
+	p2, err := nkeys.CreateCurveKeysWithRand(ef)
+	if err != nil {
+		t.Error("Failed to create key")
+		t.FailNow()
+	}
+	p1_seed, _ := p1.Seed()
+	p2_seed, _ := p2.Seed()
+
+	// Setup all the test files
+	p1_key, err := os.Create(filepath.Join(os.TempDir(), "p1_seed"))
```

Comment:
better to use `t.TempDir()` rather than `os.TempDir()` which might leave stuff behind and cause future tests to maybe act weird

Reply:
TIL `t.TempDir` was a thing...thanks!!